src/content/docs/extras.mdx

```diff
@@ -320,6 +320,215 @@ str_detect("Hello world", "ello")
 </div>
 
   
+## arrow and duckdb
+
+_Fast data storage and database_
+
+One advantage of the learning the `tidyverse` syntax is that it plugs 
+in seemlessly with databases using the 
+[dbplyr](https://dbplyr.tidyverse.org/) 
+package. The **arrow** package provides a really fast and memory efficient data 
+storage format (`.parquet`). The **duckdb** package provides a really fast and 
```

Comment:
This is a common mistake, but these are actually two separate things (albeit that they play very well together by design). Parquet defines an efficient *on-disk storage* format, while arrow defines an efficient *in-memory representation* system. Both leverage columnar representation for fast access, subsetting, reduction of cahce misses etc.. Which adds tot he confusion. (See Danielle Navaro's blog post [here](https://blog.djnavarro.net/posts/2021-11-19_starting-apache-arrow-in-r/#i-hate-you), esp. footnote 1.)

One other thing: DuckDB actually has it's own memory representation system, but it is also columnar-based and thus plays nicely with arrow's and allows the zero-copy behaviour. (In contrast, something like polars is actually built directly on top of the arrow memory model.) We don't need to go into that kind of detail here obvs, but just fwiw.

Reply:
Should add: arrow's own computation engine is called "acero". This is what you're implicitly using below when you perform summarise etc. operations below without passing to `to_duckdb()` first. But again, and somewhat confusingly, acero is separate from the memory representation standard.

I know that a lot of people show of acero first, since it's baked into the arrow package and thus comes "for free". But one reason I tend not to do that myself is that it only supports a subset of the operations that duckdb does. It's also slower. So I prefer going directly to duckdb to avoid any confusion about why, say, window functions or pivots don't work (since acero doesn't support them yet).